clientv3/integration/dial_test.go

```diff
@@ -66,6 +70,89 @@ func TestDialTLSExpired(t *testing.T) {
 	}
 }
 
+// TestDialTLSExpiredReload ensures server reloads expired certs,
+// rejecting client requests, and vice versa.
+func TestDialTLSExpiredReload(t *testing.T) {
+	defer testutil.AfterTest(t)
+
+	ts, err := copyTLSFiles(testTLSInfo)
+	if err != nil {
+		t.Fatal(err)
+	}
+	certsDir := filepath.Dir(ts.KeyFile)
+	defer os.RemoveAll(certsDir)
+
+	tse, err := copyTLSFiles(testTLSInfoExpired)
+	if err != nil {
+		t.Fatal(err)
+	}
+	dir2 := filepath.Dir(tse.KeyFile)
+	defer os.RemoveAll(dir2)
+
+	var tmpDir string
+	tmpDir, err = ioutil.TempDir(os.TempDir(), "fixtures")
+	if err != nil {
+		t.Fatal(err)
+	}
+	os.RemoveAll(tmpDir)
+	defer os.RemoveAll(tmpDir)
+
+	// start with valid certs
+	clus := integration.NewClusterV3(t, &integration.ClusterConfig{Size: 1, PeerTLS: &ts, ClientTLS: &ts})
+	defer clus.Terminate(t)
+
+	// replace certs directory with expired ones
+	if err = os.Rename(certsDir, tmpDir); err != nil {
```

Comment:
Two separate tests: atomic directory rename and copy-over. The copy-over case would cause connection failures during cert update (or might take down the server in some way).